### DISCUSSION TITLE
getargspec depreciated in latest python

### DISCUSSION THREAD

**Discussion by Istanbul348** [jfjlaros/arduino-simple-rpc#27] [General]
I'm unable to use this library with python 3.12.3. I believe it's because getargspec has been dropped now in favour of getfullargspec. I'm unable to simply change it in the init and I'm too inexperienced to understand why that doesn't work. I'd prefer not to use an older version of python if possible. 

Any chance you could either explain it or update it so that it works with python 3.12.3 please?

**Comment by jfjlaros** [comment]
Can you try the following:

    pip install setuptools

and see whether this solves the problem?

**Reply by Istanbul348** [reply]
When i enter this into a terminal I get:

Requirement already satisfied: setuptools in c:\users\****\appdata\local\programs\python\python312\lib\site-packages (70.0.0)

And it doesn't not resolve error of:
ImportError: cannot import name 'getargspec' from 'inspect'

**Comment by jfjlaros** [comment]
Strange. This package does not use the inspect library, at least not directly.

Can you share the command line invocation and the full error message?
